services/newsfeed/tests/unit/twitter.test.js

```diff
@@ -0,0 +1,144 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const {
+	safeRef,
+	tweetUrl,
+	getImageUrl,
+	tweetMapper,
+} = require('../../shared/twitter');
+
+const {
+	tweetObject,
+	retweetObject,
+	mediaTweetObject,
+	otherTweetObject,
+} = require('../constants/newsfeed');
+
+describe('Test normalizers', () => {
+	it('Test safeRef - tweet', async () => {
+		const url = 'entities.urls.0.url';
+		const result = safeRef(tweetObject, url);
+		expect(result).toBe(tweetObject.entities.urls[0].url);
+	});
+
+	it('Test safeRef - retweet', async () => {
+		const url = 'retweeted_status.entities.urls.0.url';
+		const result = safeRef(retweetObject, url);
+		expect(result).toBe(retweetObject.retweeted_status.entities.urls[0].url);
+	});
+
+	it('Test safeRef - mediaTweet', async () => {
+		const url = 'extended_entities.media.0.url';
+		const result = safeRef(mediaTweetObject, url);
+		expect(result).toBe(mediaTweetObject.extended_entities.media[0].url);
+	});
+
+	it('Test tweetUrl - tweet', async () => {
+		const url = tweetUrl(tweetObject);
+		expect(url).toBe(tweetObject.entities.urls[0].url);
+	});
+
+	it('Test tweetUrl - retweet', async () => {
+		const url = tweetUrl(retweetObject);
+		expect(url).toBe(retweetObject.retweeted_status.entities.urls[0].url);
+	});
+
+	it('Test tweetUrl - mediaTweet', async () => {
+		const url = tweetUrl(mediaTweetObject);
+		expect(url).toBe(mediaTweetObject.extended_entities.media[0].url);
+	});
+
+	it('Test tweetUrl - otherTweet', async () => {
+		const url = tweetUrl(otherTweetObject);
+		expect(url).toBe(`https://twitter.com/i/web/status/${otherTweetObject.id_str}`);
+	});
+
+	it('Test getImageUrl - tweet', async () => {
+		const url = getImageUrl(tweetObject);
+		expect(url).toBe(undefined);
+	});
+
+	it('Test getImageUrl - retweet', async () => {
+		const url = getImageUrl(tweetObject);
+		expect(url).toBe(undefined);
+	});
+
+	it('Test getImageUrl - mediaTweet', async () => {
+		const url = getImageUrl(mediaTweetObject);
+		expect(url).toBe(mediaTweetObject.entities.media[0].media_url_https);
+	});
+
+	it('Test getImageUrl - otherTweet', async () => {
+		const url = getImageUrl(otherTweetObject);
+		expect(url).toBe(undefined);
+	});
+
+	it('Test tweetMapper - tweet', async () => {
+		const mappedTweet = tweetMapper(tweetObject);
+		expect(mappedTweet).toHaveProperty('url');
+		expect(mappedTweet).toHaveProperty('image_url');
+		expect(mappedTweet).toHaveProperty('author');
```

Comment:
I think, there is no need of using `toHaveProperty` here because `objectContaining` take cares of checking if given property exists or not in response object
```suggestion
		expect(mappedTweet).toHaveProperty('author');
		expect(mappedTweet).toEqual(
			expect.objectContaining({
```

Reply:
I'd left them as redundant checks on purpose because initially, the object within `objectContaining` didn't have the `image_url` property defined within. Have removed the redundancy now.